core/model/src/main/java/com/mifos/mobilewallet/model/domain/Transaction.kt

```diff
@@ -0,0 +1,69 @@
+package com.mifos.mobilewallet.model.domain
+
+import android.os.Parcel
+import android.os.Parcelable
+import com.mifos.mobilewallet.model.domain.Currency
+import com.mifos.mobilewallet.model.domain.TransactionType
+import com.mifos.mobilewallet.model.entity.accounts.savings.TransferDetail
+
+data class Transaction(
+    var transactionId: String? = null,
+    var clientId: Long = 0,
+    var accountId: Long = 0,
+    var amount: Double = 0.0,
+    var date: String? = null,
+    var currency: Currency = Currency(),
+    var transactionType: TransactionType = TransactionType.OTHER,
+    var transferId: Long = 0,
+    var transferDetail: TransferDetail = TransferDetail(),
+    var receiptId: String? = null
+) : Parcelable {
+    constructor(parcel: Parcel) : this(
```

Comment:
Please go through the `Parcelize` annotation again. it does everything that's why we are using it and we don't it override the constructor and writeToParcel and other Parcelable function anymore. please fix in other classes as well.